elbepack/initvmaction.py

```diff
@@ -174,7 +227,37 @@ def _attach_disk_fds(self):
                 os.open(source.attrib['file'], flags),
             ])
 
-    def execute(self, _initvmdir, _opt, _args):
+    def _run_qemu_vm(self, initvmdir):
+        """
+        Start the initvm in QEMU mode.
+
+        This method starts the initvm in QEMU mode if another initvm is 
+        not already running.
+        """
+        initvmdir = check_initvm_dir(initvmdir)
+        
+        # Test if there is already a process bound to the expected port.
+        if is_soap_port_reachable():
+            if os.path.exists(os.path.join(initvmdir, 'qemu-monitor-socket')):
+                # If the unix socket exists, assume this VM is bound to the soap port.
+                print('This initvm is already running.')
+            else:
+                # If no unix socket file is found, assume another VM is bound to the soap port.
+                print('There is already another running initvm.\nPlease stop this VM first.')
+                sys.exit(211)
+        else:
+            # Try to start the QEMU VM for the given directory.
+            try:
+                subprocess.Popen(['make', 'run_qemu'], cwd=initvmdir)
```

Comment:
`subprocess.run` for consistency please. Also needs a `check=True`.

Reply:
Popen doesn't support check. It runs the command as detached process in the background, and doesn't wait for a result. This is necessary in this case, since it's the QEMU process.

Reply:
Good point, thanks for the clarification.